### PR TITLE
Added RegTestContainer SPV testing functionnalities

### LINKED ISSUE #218: @defichain/testcontainers spv testing
#### What would you like to be added:

Currently, `RegTestContainer` offers a variety of testing functions and boilerplate to ease testing. E.g. `container.generate(1)` and `container.waitForCoinBaseMaturity()`. We need to add `spv` testing functionality into RegTestContainer as well. 

1. `spv_fundaddress` as `container.spv.fundAddress()`
2. `spv_setlastheight` as `container.spv.setLastHeight()`

```ts
class RegTestContainer {
  readyonly spv = new Spv(this)
}
```

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Adds SPV testing functionnalities

#### Which issue(s) does this PR fixes?:

Fixes #218

### CODE REVIEW

**Review by fuxingloh — DISMISSED** [review]

File: packages/testcontainers/src/chains/reg_test_container/spv.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export class Spv {
+  private readonly client: RegTestContainer
+
+  constructor (client: RegTestContainer) {
+    this.client = client
+  }
+
+  /**
+   * Funds a Bitcoin address with 1 BTC.
```
Comment (surangap) [inline packages/testcontainers/src/chains/reg_test_container/spv.ts:14]: maybe mentioning this as "testing purpose only" is a good idea?
```suggestion
   * Funds a Bitcoin address with 1 BTC(for test purposes only)
```

File: packages/testcontainers/src/chains/reg_test_container/spv.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { RegTestContainer } from './'
+
+/**
+ * SPV testing functionnalities
```
Comment (surangap) [inline packages/testcontainers/src/chains/reg_test_container/spv.ts:4]: I guess you are going to add real(not only testing) `spv` RPCs also?
```suggestion
 * SPV RPCs for Bitcoin blockchain
```

**Review by surangap — CHANGES_REQUESTED** [review]
some docs improvements

File: packages/testcontainers/src/chains/reg_test_container/spv.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export class Spv {
+  private readonly client: RegTestContainer
+
+  constructor (client: RegTestContainer) {
+    this.client = client
```
Comment (canonbrother) [inline packages/testcontainers/src/chains/reg_test_container/spv.ts:10]:
```suggestion
  constructor (client: RegTestContainer) {
```

File: packages/testcontainers/src/chains/reg_test_container/spv.ts (hunk 1 above)
Comment (canonbrother) [inline packages/testcontainers/src/chains/reg_test_container/spv.ts:14]: hmm.. do we hardcode the funding amount?? @surangap  :thinking:

Comment (surangap) [reply packages/testcontainers/src/chains/reg_test_container/spv.ts:14]: looks like it.
C++ side https://github.com/DeFiCh/ain/blob/1cdc7e05b8340fbd51ba76a1263a6e1fb194e6cd/src/spv/spv_wrapper.cpp#L1586

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**